### DISCUSSION TITLE
AttestationObject cannot be deserialized

### DISCUSSION THREAD

**Discussion by dscreve** [webauthn4j/webauthn4j-spring-security#1561] [Q&A]
AttestationObject cannot be deserialized from JSON which make CredentialRecordImpl not deserializable.

Please provide an example of serialization/deserialization of CredentialRecordImpl.

com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `com.webauthn4j.data.attestation.authenticator.AuthenticatorData` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)

**Comment by ynojima** [comment]
[CredentialRecordEntity](https://github.com/webauthn4j/webauthn4j-spring-security/blob/master/samples/spa/src/main/java/com/webauthn4j/springframework/security/webauthn/sample/domain/entity/CredentialRecordEntity.java) of the sample application could be a example. It is a JPA entity class, and its fields are serialized/deserialized with [these JPA converters](https://github.com/webauthn4j/webauthn4j-spring-security/tree/master/samples/spa/src/main/java/com/webauthn4j/springframework/security/webauthn/sample/infrastructure/util/jpa/converter).
